File: repeated_string.py
def repeatedString(s, n):
    total = 0
    a = 0
    l = len(s)
    for c in s:
        if c == 'a':
            a += 1
    
    print(a)


repeatedString('abcac', 10)
# abcacabcac
# abcacabcacab

File: php/warmup/repeated_string.php
<?php

function repeatedString($s, $n){
    $l = strlen($s);
    $q = intdiv($n, $l);
    $r = $n % $l;
    $a = 0;
    
    foreach(str_split($s) as $c){
        if($c == 'a')
        $a += 1;
    }
    
    $total = $a * $q;
    
    if( $r > 0 ){
        foreach( range(1, $r) as $key => $x ){

            if($s[$key] == 'a')
                $total += 1;
        }
    }
    
    return $total;
}

print(repeatedString('a', 1000000000000));

// aba aba aba a